examples/controls_example/public/react_controls/timeslider_control/get_timeslider_control_factor.test.tsx

```diff
@@ -0,0 +1,228 @@
+ * Side Public License, v 1.
+ */
+
+import React from 'react';
+import { render, fireEvent } from '@testing-library/react';
+import { TimeRange } from '@kbn/es-query';
+import { StateComparators } from '@kbn/presentation-publishing';
+import { coreMock } from '@kbn/core/public/mocks';
+import { dataPluginMock } from '@kbn/data-plugin/public/mocks';
+import dateMath from '@kbn/datemath';
+import { BehaviorSubject } from 'rxjs';
+import { ControlGroupApi } from '../control_group/types';
+import { ControlApiRegistration } from '../types';
+import { getTimesliderControlFactory } from './get_timeslider_control_factory';
+import { TimesliderControlApi, TimesliderControlState } from './types';
+
+describe('TimesliderControlApi', () => {
+  const uuid = 'myControl1';
+  const dashboardApi = {
+    timeRange$: new BehaviorSubject<TimeRange | undefined>(undefined),
+  };
+  const controlGroupApi = {
+    autoApplySelections$: new BehaviorSubject(true),
+    parentApi: dashboardApi,
+  } as unknown as ControlGroupApi;
+  const dataStartServiceMock = dataPluginMock.createStartContract();
+  dataStartServiceMock.query.timefilter.timefilter.calculateBounds = (timeRange: TimeRange) => {
+    const now = new Date();
+    return {
+      min: dateMath.parse(timeRange.from, { forceNow: now }),
+      max: dateMath.parse(timeRange.to, { roundUp: true, forceNow: now }),
+    };
+  };
+  const factory = getTimesliderControlFactory({
+    core: coreMock.createStart(),
+    data: dataStartServiceMock,
+  });
+  let comparitors: StateComparators<TimesliderControlState> | undefined;
+  function buildApiMock(
+    api: ControlApiRegistration<TimesliderControlApi>,
+    nextComparitors: StateComparators<TimesliderControlState>
+  ) {
+    comparitors = nextComparitors;
+    return {
+      ...api,
+      uuid,
+      parentApi: controlGroupApi,
+      unsavedChanges: new BehaviorSubject<Partial<TimesliderControlState> | undefined>(undefined),
+      resetUnsavedChanges: () => {},
+      type: factory.type,
+    };
+  }
+
+  beforeEach(() => {
+    dashboardApi.timeRange$.next({
+      from: '2024-06-09T00:00:00.000Z',
+      to: '2024-06-10T00:00:00.000Z',
+    });
+  });
+
+  test('Should set timeslice to undefined when state does not provide percentage of timeRange', () => {
+    const { api } = factory.buildControl({}, buildApiMock, uuid, controlGroupApi);
+    expect(api.timeslice$.value).toBe(undefined);
+  });
+
+  test('Should set timeslice to values within time range when state provides percentage of timeRange', () => {
+    const { api } = factory.buildControl(
+      {
+        timesliceStartAsPercentageOfTimeRange: 0.25,
+        timesliceEndAsPercentageOfTimeRange: 0.5,
+      },
+      buildApiMock,
+      uuid,
+      controlGroupApi
+    );
+
+    expect('2024-06-09T06:00:00.000Z').toEqual(new Date(api.timeslice$.value![0]).toISOString());
+    expect('2024-06-09T12:00:00.000Z').toEqual(new Date(api.timeslice$.value![1]).toISOString());
```

Comment:
**very tiny nit:** I just personally find it easier to read the other way around 😅 

```suggestion
    expect(new Date(api.timeslice$.value![0]).toISOString()).toEqual('2024-06-09T06:00:00.000Z');
    expect(new Date(api.timeslice$.value![1]).toISOString()).toEqual('2024-06-09T12:00:00.000Z');
```

This applies to all of the checks in the other tests, too - but just leaving a single comment here since it's definitely a stylistic thing.